rds-update-rdsh-collection/azuredeploy.json

```diff
@@ -0,0 +1,318 @@
+            },
+            "defaultValue": "https://raw.githubusercontent.com/azure/azure-quickstart-templates/master/rds-update-rdsh-collection/"
+        },
+        "_artifactsLocationSasToken": {
+            "type": "securestring",
+            "metadata": {
+                "description": "The sasToken required to access _artifactsLocation.  When the template is deployed using the accompanying scripts, a sasToken will be automatically generated."
+            },
+            "defaultValue": ""
+        }
+
+    },
+
+    "variables": {
+
+        "iteration": "[toLower(parameters('rdshUpdateIteration'))]",
+
+        "assetLocation": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/rds-deployment/",
+
+        "storageAccountName": "[split( split( parameters('rdshTemplateImageUri'), '/')[2], '.' )[0]]",
+
+        "impersonateScript": "https://gallery.technet.microsoft.com/scriptcenter/Impersonate-a-User-9bfeff82/file/127189/1/New-ImpersonateUser.ps1",
+
+        "p": {
+            "domain": "[concat(' -domain ', parameters('existingDomainName'))]",
+            "user": "[concat(' -username ', parameters('existingAdminUsername'))]",
+            "pwd": "[concat(' -password ', parameters('existingAdminPassword'))]",
+            "collection": "[concat(' -collection \"', parameters('existingRdshCollectionName'), '\"')]",
+            "iteration": "[concat(' -iteration ', toLower(parameters('rdshUpdateIteration')))]",
+            "servers": "[concat(' -nServers ', parameters('rdshNumberOfInstances'))]",
+            "timeout": "[concat(' -nTimeoutMinutes ', parameters('userLogoffTimeoutInMinutes'))]"
+        },
+        "scriptParameters": "[concat(variables('p').domain, variables('p').user, variables('p').pwd, variables('p').collection, variables('p').iteration, variables('p').servers, variables('p').timeout)]",
+        "scriptFolder": "Scripts",
+        "scriptFileName":  "Script.ps1"
+    },
+
+    "resources": [
+
+        {
+            "apiVersion": "2015-06-15",
+            "type": "Microsoft.Network/networkInterfaces",
+            "name": "[concat('rdsh-', copyindex(),variables('iteration'), '-nic')]",
+            "location": "[resourceGroup().location]",
+            "copy": {
+                "name": "rdsh-nic-loop",
+                "count": "[parameters('rdshNumberOfInstances')]"
+            },
+            "properties": {
+                "ipConfigurations": [
+                    {
+                        "name": "ipconfig",
+                        "properties": {
+                            "privateIPAllocationMethod": "Dynamic",
+                            "subnet": {
+                                "id": "[concat(resourceId('Microsoft.Network/virtualNetworks', parameters('existingVnetName')), '/subnets/', parameters('existingSubnetName'))]"
```

Comment:
simplify just use resourceId()

resourceId('Microsoft.Network/virtualNetworks/subnets', parameters('existingVnetName')), parameters('existingSubnetName'))

include the resourcegroup param as appropriate